src/Microsoft.AspNetCore.Server.Kestrel/Http/FrameRequestStream.cs

```diff
@@ -165,13 +165,13 @@ private Task<int> ValidateState(CancellationToken cancellationToken)
                 case FrameStreamState.Open:
                     if (cancellationToken.IsCancellationRequested)
                     {
-                        return TaskUtilities.GetCancelledZeroTask();
+                        return TaskUtilities.GetCancelledZeroTask(cancellationToken);
                     }
                     break;
                 case FrameStreamState.Closed:
                     throw new ObjectDisposedException(nameof(FrameRequestStream));
                 case FrameStreamState.Aborted:
-                    return TaskUtilities.GetCancelledZeroTask();
+                    return TaskUtilities.GetCancelledZeroTask(default(CancellationToken));
```

Comment:
Is by design https://github.com/aspnet/KestrelHttpServer/pull/567#discussion_r50049886 and the reading and writing are different.